sync/sync_head.go

```diff
@@ -49,7 +47,7 @@ func (s *Syncer[H]) Head(ctx context.Context) (H, error) {
 	// NOTE: We could trust the netHead like we do during 'automatic subjective initialization'
 	// but in this case our subjective head is not expired, so we should verify netHead
 	// and only if it is valid, set it as new head
-	s.incomingNetworkHead(ctx, netHead)
+	_ = s.incomingNetworkHead(ctx, netHead)
```

Comment:
in the spirit of hardening, it might be worth at least logging the error for context.

Reply:
It's logged inside `incomingNetworkHead`, but we may mention here that failure us coming from the trustedpeer

Reply:
this getter here (after my PR) could also request tracked peers so maybe it's at least worth logging as a debug?

Reply:
Your change will minimize the chances we request trustedPeers, meaning that this `incomingNetworkHead` call is now similar to when we receive a gossiped header. The gossiping layer does not log failure, while we do that inside of `incomingNetworkHead`, so adding a log, considering your change, is even more redundant.

Reply:
Actually, there is no way to know if we got an error from a trusted peer or not after #53

Reply:
When we request recent, it can be anyone and the error does not say us who. The exchange could also log that as well.